src/test/java/org/opentripplanner/middleware/testutils/OtpMiddlewareTestEnvironment.java

```diff
@@ -0,0 +1,60 @@
+package org.opentripplanner.middleware.testutils;
+
+import org.opentripplanner.middleware.OtpMiddlewareMain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.isRunningCi;
+
+/**
+ * This class contains various fields pertaining to the test environment and also contains a static block of code that
+ * starts an OtpMiddleware server (which also loads the config and initializes the database). Every test class that
+ * requires a test instance of the otp-middleware server to be running must extend this class. If test classes that
+ * extend this class have additional setup logic that needs to be performed, they can add their own setup method with
+ * the @BeforeAll decorator which will be run after the static code in the class is invoked.
+ */
+public class OtpMiddlewareTestEnvironment {
+    private static final Logger LOG = LoggerFactory.getLogger(OtpMiddlewareTestEnvironment.class);
+
+    /**
+     * Whether the end-to-end environment variable is enabled.
+     */
+    public static final boolean IS_END_TO_END = CommonTestUtils.getBooleanEnvVar("RUN_E2E");
+
+    static final String TEST_RESOURCE_PATH = "src/test/resources/org/opentripplanner/middleware/";
+
+    static {
```

Comment:
Any reason we're statically initializing and not just using `setup` with `@BeforeAll`?

Reply:
If this logic is placed in a static method with the `@BeforeAll` decorator, it'll be ran every time another class begins its setup. We could have some static variable that gets updated like before, of just use this static block that only runs once. I think it's simpler this way.